app/controllers/integrations_controller.rb

```diff
@@ -44,5 +39,14 @@ def set_project
     @project = policy_scope(Project).friendly.find(params[:project_id])
   end
 
+  def set_integrations
+    @integrations ||= begin
+      current_integrations = @project.integrations
+      valid_integrations = Integration::VALID_INTEGRATIONS
+      missing_integrations = valid_integrations - current_integrations.pluck(:kind)
+
+      current_integrations |=  missing_integrations.map { |i| Integration.new kind: i }
```

Comment:
I think that you don't need the assigner `|=` here, you can use just `+`.

Also, I would remove variable `valid_integrations` in line 45 and use `Integration::VALID_INTEGRATIONS` directly in 46, but this is just a suggestion to remove extra lines.